server/src/middleware/auth.ts

```diff
@@ -8,12 +8,18 @@ import { getUserBySessionId } from '../lib/db';
  * Sets req.user to undefined if no user is logged in
  */
 const authMiddleware = async (req: Request, res: Response, next: Function) => {
-    const { sessionId } = req.cookies;
+    let { sessionId } = req.cookies;
 
-    // Check for missing parameters
     if (sessionId === undefined || typeof sessionId !== 'string') {
-        req.user = undefined;
-        return next();
+        // check for Auth header (used by plugin) - format is "dojo <sessionId>"
+        const authHeader = req.headers.authorization;
+        if (authHeader && authHeader.startsWith('dojo ')) {
+            [, sessionId] = authHeader.split(' ');
```

Comment:
It might be cleaner to do this before the `if (sessionId === undefined)` case. As the sessionId can be undefined if an invalid header is passed along. It will be fine since no valid user can be found for this session. But it might be cleaner to do it before, since there is a check for undefined sessionIds anyway

Reply:
I'm not really following "As the sessionId can be undefined if an invalid header is passed along" - but I'm not super happy with this part anyway, so let me just rewrite it and maybe we mean the same thing 👍